File: app/src/main/java/com/test/shoppingapp/shoppinglist/di/ShoppingListModule.java
package com.test.shoppingapp.shoppinglist.di;

import android.support.v7.app.AppCompatActivity;

import com.test.shoppingapp.shoppinglist.ShoppingListActivity;

import dagger.Module;
import dagger.Provides;

@Module
public class ShoppingListModule {

    @Provides
    public AppCompatActivity provideShoppingListActivity(ShoppingListActivity shoppingListActivity) {
        return shoppingListActivity;
    }
}

File: app/src/main/java/com/test/shoppingapp/shoppinglist/di/ShoppingListModule.kt
package com.test.shoppingapp.shoppinglist.di

import androidx.appcompat.app.AppCompatActivity

import com.test.shoppingapp.shoppinglist.ShoppingListActivity

import dagger.Module
import dagger.Provides

@Module
class ShoppingListModule {

    @Provides
    fun provideShoppingListActivity(shoppingListActivity: ShoppingListActivity): AppCompatActivity {
        return shoppingListActivity
    }
}
